src/sparse/indptr.rs

```diff
@@ -381,6 +381,19 @@ impl<Iptr: SpIndex> IndPtr<Iptr> {
             *val += Iptr::one();
         }
     }
+
+    /// Slice this indptr to include only the outer dimensions in the range
+    /// `start..end`.
+    pub(crate) fn middle_slice(
```

Comment:
This method should be implemented on `IndPtrBase`

Reply:
That's the same debate as the previous point, if I implement it on `IndPtrBase` I need to take another name for the "reborrowing" version. I wish it was possible to "specialize" on this lifetime, or perform compile-time lifetime computations to figure out the correct lifetime...

Reply:
It's now implemented on `IndPtrBase` with the demonomorphization trick of https://github.com/vbarrielle/sprs/pull/252#discussion_r540551472